src/util/CollectionUtils.tsx

```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
     const id = link.substring(link.lastIndexOf("/") + 1, link.indexOf("?"));
     return id[0] ?? "";
   }
+
+  static getAuthor(author: string): JSX.Element {
+    if (!author) return null;
+    const authorLinkText = author;
+    const query: ApiSearchQuery = {
+      query: `author:${author}`,
+    };
+    return (
+      <Link
+        to={{
+          pathname: "/search",
+          query: query,
+        }}
+        className="link"
+      >
+        {authorLinkText}
+      </Link>
+    );
+  }
+
+  /**
+   * Get publisher and publish location
+   * @param pubPlace - The display name of the place of publication
+   * @param agents - an array of Agents
+   * @returns A display element for publisher and location
+   */
+  static getPublisherAndLocation(
+    pubPlace: string,
+    publisher: string
+  ): JSX.Element {
+    const publisherDisplayLocation = (pubPlace: string) => {
+      return pubPlace
+        ? ` in ${truncateStringOnWhitespace(pubPlace, MAX_PLACE_LENGTH)}`
+        : "";
+    };
+
+    const publisherDisplayText = (publisher: string) => {
+      if (!publisher) return "";
+      return ` by ${truncateStringOnWhitespace(
+        publisher,
+        MAX_PUBLISHER_NAME_LENGTH
+      )}`;
+    };
+
+    const displayLocation = publisherDisplayLocation(pubPlace);
+    const displayName = publisherDisplayText(publisher);
+    if (!displayLocation && !displayName)
+      return <>Publisher and Location Unknown</>;
+    const publisherText = `Published${displayLocation}${displayName}`;
+    return <>{publisherText}</>;
+  }
+
+  // Language Display
+  static getLanguageDisplayText(language: string) {
+    if (language) {
+      const languageText = `Languages: ${language}`;
+      return <>{languageText}</>;
+    }
+    return <>Languages: Undetermined</>;
+  }
+
+  // Rights
+  static getLicense(rights: Rights) {
+    return rights ? `License: ${rights.rightsStatement}` : "License: Unknown";
+  }
+
+  static getReadLink = (links: OpdsLink[], type: "readable" | "embedable") => {
```

Comment:
Since it seems like the `type: "readable" | "embedable"` is specific to what we can read/open in this app, we might want to make is a named type that we can reuse. Something like `export type ReadOnlineTypes = "readable" | "embedable"`